x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -176,16 +179,19 @@ const IntegrationPostureDashboard = ({
 
 export const ComplianceDashboard = () => {
   const [selectedTab, setSelectedTab] = useState(CSPM_POLICY_TEMPLATE);
-  const getSetupStatus = useCspSetupStatusApi();
-  const hasFindingsKspm =
-    getSetupStatus.data?.kspm?.status === 'indexed' ||
-    getSetupStatus.data?.indicesDetails[0].status === 'not-empty';
-  const hasFindingsCspm =
-    getSetupStatus.data?.cspm?.status === 'indexed' ||
-    getSetupStatus.data?.indicesDetails[0].status === 'not-empty';
+  const { data: getSetupStatus } = useCspSetupStatusApi();
+  const status = getCpmStatus(getSetupStatus);
+  const hasFindingsKspm = status.hasKspmFindings;
+  const hasFindingsCspm = status.hasCspmFindings;
   const cspmIntegrationLink = useCspIntegrationLink(CSPM_POLICY_TEMPLATE);
   const kspmIntegrationLink = useCspIntegrationLink(KSPM_POLICY_TEMPLATE);
 
+  const isKspmInstalled = status.isKspmInstalled;
+  const isCspmInstalled = status.isCspmInstalled;
+
+  const isCspmIntegrationInstalled = status.isCspmIntegrationInstalled;
+  const isKspmIntegrationInstalled = status.isKspmIntegrationInstalled;
+
```

Comment:
No need to declare all that variables, you can get by destructuring the getCpmStatus function

```js
const {isKspmInstalled, isCspmInstalled, isCspmIntegrationInstalled, isKspmIntegrationInstalled} = getCpmStatus(getSetupStatus);
```